druid/src/widget/flex.rs

```diff
@@ -225,15 +271,21 @@ impl<T: Data> Widget<T> for Flex<T> {
                 let min_major = if major.is_infinite() { 0.0 } else { major };
                 let child_bc = self.direction.constraints(&loosened_bc, min_major, major);
                 let child_size = child.widget.layout(layout_ctx, &child_bc, data, env);
+                flex_used += self.direction.major(child_size);
                 minor = minor.max(self.direction.minor(child_size));
                 // Stash size.
                 let rect = Rect::from_origin_size(Point::ORIGIN, child_size);
                 child.widget.set_layout_rect(rect);
             }
         }
 
+        // figure out if we have extra space on major axis, and if so how to use it
+        let major_used = total_non_flex + flex_used;
+        let extra = (self.direction.major(bc.min()) - major_used).max(0.0);
+        let spacing = self.main_alignment.spacing(extra, self.children.len());
+
         // Finalize layout, assigning positions to each child.
-        let mut major = 0.0;
+        let mut major = spacing.pre;
```

Comment:
I don't know if there should be a warning here or really anything different at all, I just know I wouldn't feel comfortable modifying this section. My brain compiler is tracking at like 85%.

Reply:
so 'major' here represents our 'current' position in the major axis; that is, where we're going to place the next item. This means that we should place the first item after any initial padding.